### DISCUSSION TITLE
How to get the IP

### DISCUSSION THREAD

**Discussion by LapsTimeOFF** [honojs/hono#1187] [Q&A]
Hi, for one of my request, I need to get the IP of the user, but I cant find it in the docs

**Comment by yusukebe** [comment]
Hono only deals with Web-standard APIs, so it does not need to provide a method for users to retrieve IP addresses. Therefore, the ability to get an IP address depends on the platform, not Hono's core. For example, on Cloudflare Workers, you can get `country` or `city` from `Request.cf`, but retrieving IP is not available:

https://developers.cloudflare.com/workers/runtime-apis/request/#incomingrequestcfproperties

For Node.js, we have our own node server adapter, so perhaps, we may pass the IP address to a Hono app.
